src/stk_mcp/stk_logic/core.py

```
@@ -1,95 +1,124 @@
-        # Try attaching to existing STK
-        print("   Attempting to attach to existing STK instance...")
-        state.stk_ui_app = STKDesktop.AttachToApplication()
-        state.stk_root = state.stk_ui_app.Root
-        print("   Successfully attached to existing STK instance.")
-        state.stk_ui_app.Visible = True
-        # Close any open scenario to start clean
-        if state.stk_root and state.stk_root.Children.Count > 0:
-             print(f"   Closing existing scenario '{state.stk_root.CurrentScenario.InstanceName}'...")
-             state.stk_root.CloseScenario()
+    @asynccontextmanager
+    async def stk_lifespan_manager(server: FastMCP) -> AsyncIterator[StkState]:
```

Ruff (0.12.2)

Unused function argument: `server`

(ARG001)